diff --git a/node_modules/react-native-svg/.bun-tag-55c72d773cdb59fb b/.bun-tag-55c72d773cdb59fb
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/node_modules/react-native-svg/.bun-tag-f522b4a5929fac99 b/.bun-tag-f522b4a5929fac99
new file mode 100644
index 0000000000000000000000000000000000000000..e69de29bb2d1d6434b8b29ae775ad8c2e48c5391
diff --git a/lib/commonjs/utils/fetchData.js b/lib/commonjs/utils/fetchData.js
index 207b2937d048ca9c22b58dd7be3641b5a7262274..d6a45c9fd9b69beaf5b3676f5702f71279c7ac84 100644
--- a/lib/commonjs/utils/fetchData.js
+++ b/lib/commonjs/utils/fetchData.js
@@ -5,7 +5,7 @@ Object.defineProperty(exports, "__esModule", {
 });
 exports.fetchText = fetchText;
 var _reactNative = require("react-native");
-var _buffer = require("buffer");
+
 async function fetchText(uri) {
   if (!uri) {
     return null;
@@ -23,7 +23,7 @@ const decodeBase64Image = uri => {
   const splitContent = decoded.split(';')[1].split(',');
   const dataType = splitContent[0];
   const content = splitContent.slice(1).join(',');
-  return _buffer.Buffer.from(content, dataType).toString('utf-8');
+  return Buffer.from(content, dataType).toString('utf-8');
 };
 function dataUriToXml(uri) {
   try {
diff --git a/lib/module/lib/extract/transform.js b/lib/module/lib/extract/transform.js
index 06c724b760232c93e5ed9d72266a054d5fe65512..760929ccd9ffdee5c27accaf6d3eefaa83f2e721 100644
--- a/lib/module/lib/extract/transform.js
+++ b/lib/module/lib/extract/transform.js
@@ -1478,8 +1478,6 @@ function peg$parse(input, options) {
     throw peg$buildStructuredError(peg$maxFailExpected, peg$maxFailPos < input.length ? input.charAt(peg$maxFailPos) : null, peg$maxFailPos < input.length ? peg$computeLocation(peg$maxFailPos, peg$maxFailPos + 1) : peg$computeLocation(peg$maxFailPos, peg$maxFailPos));
   }
 }
-module.exports = {
-  SyntaxError: peg$SyntaxError,
-  parse: peg$parse
-};
+export const SyntaxError = peg$SyntaxError;
+export const parse = peg$parse;
 //# sourceMappingURL=transform.js.map
diff --git a/lib/module/lib/extract/transformToRn.js b/lib/module/lib/extract/transformToRn.js
index e2a7f4ca95728be34c45e55b96ce5ec1f43d14cc..c496d600fc0f91246fb41acfd6fdf1363c193cc9 100644
--- a/lib/module/lib/extract/transformToRn.js
+++ b/lib/module/lib/extract/transformToRn.js
@@ -1148,9 +1148,7 @@ function peg$parse(input, options) {
     throw peg$buildStructuredError(peg$maxFailExpected, peg$maxFailPos < input.length ? input.charAt(peg$maxFailPos) : null, peg$maxFailPos < input.length ? peg$computeLocation(peg$maxFailPos, peg$maxFailPos + 1) : peg$computeLocation(peg$maxFailPos, peg$maxFailPos));
   }
 }
-module.exports = {
-  StartRules: ['start'],
-  SyntaxError: peg$SyntaxError,
-  parse: peg$parse
-};
+export const StartRules = ['start'];
+export const SyntaxError = peg$SyntaxError;
+export const parse = peg$parse;
 //# sourceMappingURL=transformToRn.js.map
diff --git a/lib/module/utils/fetchData.js b/lib/module/utils/fetchData.js
index 12b68966cbcfd9a18cae66824dfe7aec60d14a80..dda9fab51f762500eb3b1b2f84a5a6b0b1fbd4eb 100644
--- a/lib/module/utils/fetchData.js
+++ b/lib/module/utils/fetchData.js
@@ -1,5 +1,5 @@
 import { Platform } from 'react-native';
-import { Buffer } from 'buffer';
+
 export async function fetchText(uri) {
   if (!uri) {
     return null;
diff --git a/src/utils/fetchData.ts b/src/utils/fetchData.ts
index d141be31c924bd1d3ec287ad4e291db018b4bb72..0dc3d95ca382151a45af41f8ed499d1403a1f65c 100644
--- a/src/utils/fetchData.ts
+++ b/src/utils/fetchData.ts
@@ -1,5 +1,4 @@
 import { Platform } from 'react-native';
-import { Buffer } from 'buffer';
 
 export async function fetchText(uri?: string): Promise<string | null> {
   if (!uri) {
