diff --git a/src/web/generator.ts b/src/web/generator.ts
index 26db130f12f3edfe996a3cf08eef155b696a059a..b5b11ba46e8b773c4687ad77c5c5cc6c3fa5d4e0 100644
--- a/src/web/generator.ts
+++ b/src/web/generator.ts
@@ -4,7 +4,7 @@
 //
 
 import { useId, useInsertionEffect } from "react";
-import { prefix } from "inline-style-prefixer";
+import { prefix } from "inline-style-prefixer/lib";
 import { Properties as CssProperties } from "csstype";
 import { Theme, breakpoints, useTheme } from "../theme";
 import {
