diff --git a/pg-core/dialect.cjs b/pg-core/dialect.cjs
index b2c2a77086503f751985bc6928d398c62cf79762..45adac1a94a0019767111a7dd1b69d476f22a326 100644
--- a/pg-core/dialect.cjs
+++ b/pg-core/dialect.cjs
@@ -358,7 +358,14 @@ class PgDialect {
   buildInsertQuery({ table, values: valuesOrSelect, onConflict, returning, withList, select, overridingSystemValue_ }) {
     const valuesSqlList = [];
     const columns = table[import_table2.Table.Symbol.Columns];
-    const colEntries = Object.entries(columns).filter(([_, col]) => !col.shouldDisableInsert());
+    let colEntries = Object.entries(columns);
+    colEntries = select && !is(valuesOrSelect, SQL)
+        ? Object
+            .keys(valuesOrSelect._.selectedFields)
+            .map((key) => [key, columns[key]])
+        : overridingSystemValue_
+            ? colEntries
+            : colEntries.filter(([_, col]) => !col.shouldDisableInsert());
     const insertOrder = colEntries.map(
       ([, column]) => import_sql2.sql.identifier(this.casing.getColumnCasing(column))
     );
diff --git a/pg-core/dialect.js b/pg-core/dialect.js
index 31a019cc0a6bc67303a745fdf9ad613e8fbdcb5f..430cd0a0e82ec5d3938c97c17cc5d2924935c9e9 100644
--- a/pg-core/dialect.js
+++ b/pg-core/dialect.js
@@ -356,7 +356,14 @@ class PgDialect {
   buildInsertQuery({ table, values: valuesOrSelect, onConflict, returning, withList, select, overridingSystemValue_ }) {
     const valuesSqlList = [];
     const columns = table[Table.Symbol.Columns];
-    const colEntries = Object.entries(columns).filter(([_, col]) => !col.shouldDisableInsert());
+    let colEntries = Object.entries(columns);
+    colEntries = select && !is(valuesOrSelect, SQL)
+        ? Object
+        .keys(valuesOrSelect._.selectedFields)
+        .map((key) => [key, columns[key]])
+        : overridingSystemValue_
+            ? colEntries
+            : colEntries.filter(([_, col]) => !col.shouldDisableInsert());
     const insertOrder = colEntries.map(
       ([, column]) => sql.identifier(this.casing.getColumnCasing(column))
     );
diff --git a/pg-core/query-builders/insert.cjs b/pg-core/query-builders/insert.cjs
index 22e0a4b4ad7ac64b065fc540416c6ed15ff4336b..7562b76a0d8f0406385bddabd7cbaa4af03c219f 100644
--- a/pg-core/query-builders/insert.cjs
+++ b/pg-core/query-builders/insert.cjs
@@ -76,11 +76,6 @@ class PgInsertBuilder {
   }
   select(selectQuery) {
     const select = typeof selectQuery === "function" ? selectQuery(new import_query_builder.QueryBuilder()) : selectQuery;
-    if (!(0, import_entity.is)(select, import_sql.SQL) && !(0, import_utils.haveSameKeys)(this.table[import_table.Columns], select._.selectedFields)) {
-      throw new Error(
-        "Insert select error: selected fields are not the same or are in a different order compared to the table definition"
-      );
-    }
     return new PgInsertBase(this.table, select, this.session, this.dialect, this.withList, true);
   }
 }
diff --git a/pg-core/query-builders/insert.js b/pg-core/query-builders/insert.js
index 60a8bb0d1c22b890bd8fbf4c85d5df41ca42444c..638091e488275172c8513cb4e4f94aacc6ac5978 100644
--- a/pg-core/query-builders/insert.js
+++ b/pg-core/query-builders/insert.js
@@ -52,11 +52,6 @@ class PgInsertBuilder {
   }
   select(selectQuery) {
     const select = typeof selectQuery === "function" ? selectQuery(new QueryBuilder()) : selectQuery;
-    if (!is(select, SQL) && !haveSameKeys(this.table[Columns], select._.selectedFields)) {
-      throw new Error(
-        "Insert select error: selected fields are not the same or are in a different order compared to the table definition"
-      );
-    }
     return new PgInsertBase(this.table, select, this.session, this.dialect, this.withList, true);
   }
 }
